project/src/main/puzzle/piece/piece-queue.gd

```diff
@@ -24,14 +27,27 @@ var _default_piece_types := PieceTypes.default_types
 
 var _remaining_piece_count := UNLIMITED_PIECES
 
+## When line pieces are enabled, they are pulled from a 'cheat bag'.
+##
+## Imagine a bag with eight blue balls and one red ball. For each opportunity where a line piece could be added to the
+## piece queue, you first pull a ball from the bag. If it is red, then you add a line piece, otherwise you don't. Once
+## the bag is empty, it is refilled.
+##
+## These two fields correspond to the number of line pieces and total pieces remaining in the cheat bag.
+var _cheat_bag_line_pieces_remaining := 0
+var _cheat_bag_pieces_remaining := 0
+var _popped_piece_count := 0
```

Comment:
comment

Reply:
I don't think a comment is necessary as the field name is self-explanatory, but i moved the field so it's not covered by the `These two fields...` comment